### PR TITLE
Flatmanga : better chapter title handling

### PR DESCRIPTION
**Pull request by MikeZeDev**
If there is a special regex character in the title (like +, [ ] () ) , an exception was caused and no chapter was returned. Better having a long a* title than no chapter at all.

Exemple : KLMANGA > + Tic Nee Sans , [HAISIN CHUU]

### CODE REVIEW

File: src/web/mjs/connectors/templates/FlatManga.mjs [hunk 1]
```diff
@@ -58,7 +58,11 @@ export default class FlatManga extends Connector {
             }
             let title = (this.queryChapterTitle ? element.querySelector(this.queryChapterTitle) : element).textContent.replace(manga.title, '');
             let mangaTitle = manga.title.replace(/\s*-\s*RAW$/, '');
-            title = title.replace(new RegExp(mangaTitle, 'i'), '');
```
Comment (Sheepux) [inline src/web/mjs/connectors/templates/FlatManga.mjs:61]: how about `new RegExp(myregex.replace(/[#-}]/g, '\\$&'))` to sanitize the string ?

Comment (Sheepux) [reply src/web/mjs/connectors/templates/FlatManga.mjs:61]: did you test it well first ? I'm not sure if it covers all use cases

Comment (MikeZeDev) [reply src/web/mjs/connectors/templates/FlatManga.mjs:61]: Hum, if i force the manga title to be in chapter name (for testing), and in use the regex fixed, nothing is removed :/
```
            title += ' '+ mangaTitle;
            title = title.replace(new RegExp(mangaTitle.replace(/[#-}]/g, '\\$&')), '');
```
image

Gotta dig a bit

Comment (MikeZeDev) [reply src/web/mjs/connectors/templates/FlatManga.mjs:61]: OK got a better regex using this as reference https://www.debuggex.com/cheatsheet/regex/javascript

`            title = title.replace(new RegExp(mangaTitle.replace(/[\*\^\$\.\|\?\+\-\(\)\[\]\{\}]/g, '\\$&')), '');`

Seems to work fine

Comment (MikeZeDev) [reply src/web/mjs/connectors/templates/FlatManga.mjs:61]: Using this as reference : https://exlskills.com/learn-en/courses/javascript-fundamentals-basics_javascript/regular-expressions-cWSVBpppzSat/applications-lPjEnTmnZThz/special-characters-in-regex-xMIVQGkIObNm

`[\*\^\$\.\|\?\+\-\(\)\[\]\{\}]` seems to work fine. We escape only special chars and it works.

**Comment by MikeZeDev** [comment]
image

Works fine.

**Review by Sheepux — APPROVED** [review]

**Pull request merged**